debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/gccheck/CheckEngine.java

```diff
@@ -1029,7 +1029,7 @@ public int checkJ9ClassPointer(J9ClassPointer clazz, boolean allowUndead) throws
 		}
 
 		// Short circuit if we've recently checked this class.
-		int cacheIndex = (int) (clazz.longValue() % CLASS_CACHE_SIZE);
+		int cacheIndex = new UDATA(clazz.longValue()).mod(CLASS_CACHE_SIZE).intValue();
```

Comment:
A more concise expression would be `UDATA.cast(clazz).mod(CLASS_CACHE_SIZE).intValue()`.